kernel/process/process.rs

```diff
@@ -373,7 +373,10 @@ impl Process {
         if let Some(new) = set {
             let new_set = new.read::<[u8; 128]>()?;
             match how {
-                SignalMask::Block => sigset.assign_or(new_set),
+                SignalMask::Block => {
+                    sigset.assign_or(new_set);
+                    sigset.assign_and_not(EXCLUDING_SIGNAL_MASK);
```

Comment:
Please rewrite the check by an `if` statement instead of consuming 128 bytes

Reply:
Do you mean check in the try_delivering_signal? 
```rust
if signal == SIGSTOP
    || signal == SIGKILL
    || !sigset.get(signal as usize).unwrap_or(true)
```
the alternative would be to unset both of them (which is what I should have done from the start)
```rust
sigset.assign_or(new_set);
sigset.unset(SIGKILL)
sigset.unset(SIGSTOP)
```
I am asking because I do not believe that checking if those signals are set when setting the mask has any sense performance-wise. And how I understand man page we should ignore attempts to set them. Which, if I got it right, means that when we return sigset to the caller it should not contain those signals. 

But, again, I can be wrong :)

Reply:
No I mean I'd avoid adding `EXCLUDING_SIGNAL_MASK` since it's too sparse just for pattern matching. Since as you mentioned we'll handle some signals (e.g. `SIGKILL`) regardless of the process's `sigset`, I think it's safe to remove  the `sigset.assign_and_not(EXCLUDING_SIGNAL_MASK);` line.

Reply:
I think I finally got what you meant  :)